### PR TITLE
util/stmtsummary: migrate test-infra to testify (part 2)

### LINKED ISSUE #26420: migrate test-infra to testify for `util/stmtsummary` pkg

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #26420 
split from #26771, [2/2]
based on #26870  

### Release note 
```release-note
None
```

### CODE REVIEW

**Comment by feitian124** [comment]
/cc @tisonkun
hi, please help review, thanks

File: util/stmtsummary/statement_summary_test.go [hunk 1]
```diff
@@ -48,30 +42,27 @@ func fakePlanDigestGenerator() string {
...
-	err = s.ssMap.SetRefreshInterval("1800", false)
-	c.Assert(err, IsNil)
-	err = s.ssMap.SetHistorySize("24", false)
-	c.Assert(err, IsNil)
-}
-
-func TestT(t *testing.T) {
-	CustomVerboseFlag = true
-	TestingT(t)
+func TestSetUp(t *testing.T) {
```
Comment (xiongjiwei) [inline util/stmtsummary/statement_summary_test.go:45]: I got a question about this test, what is it used for? Seems that it is not setup anything? Is it used for test function like `newStmtSummaryByDigestMap`

Comment (xiongjiwei) [reply util/stmtsummary/statement_summary_test.go:45]: ok, thinks for your explanation

File: util/stmtsummary/statement_summary_test.go [hunk 2]
```diff
@@ -1008,114 +1005,116 @@ func (s *testStmtSummarySuite) TestSetMaxStmtCountParallel(c *C) {
...
-			s.ssMap.AddStatement(stmtExecInfo1)
+			ssMap.AddStatement(stmtExecInfo1)
 		}
 	}
 	for i := 0; i < threads; i++ {
 		go addStmtFunc()
 	}
 
-	defer c.Assert(s.ssMap.SetMaxStmtCount("", true), IsNil)
+	defer require.Nil(t, ssMap.SetMaxStmtCount("", true))
```
Comment (xhebox) [inline util/stmtsummary/statement_summary_test.go:1015]: Arguments are evaluated immediately, maybe the original logic is wrong? I guess he want to `setmaxstmtcount` after the return.

Comment (xhebox) [reply util/stmtsummary/statement_summary_test.go:1015]: Oops, it is in the part1, I've commented in 26870, resolved.

**Review by tisonkun — COMMENTED** [review]
@feitian124 plz ping me when part 1 is merged and you squash the diff.

**Comment by feitian124** [comment]
> @feitian124 plz ping me when part 1 is merged and you squash the diff.

ok

**Comment by feitian124** [comment]
/cc @tisonkun 
part0, part1 is merged, please check this last one

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by xhebox — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**